content/software/ide-v2/tutorials/ide-v2-customize-auto-formatter/content.md

```diff
@@ -0,0 +1,221 @@
+---
+title: 'Customizing the Auto Formatter Feature'
+difficulty: easy
+description: 'Learn how to configure the auto formatter feature'
+tags:
+ - Auto Formatter
+ - Tools
+author: 'Benjamin Dannegård, Per Tillisch'
+---
+
+Selecting **Tools > Auto Format** or pressing CTRL + T on Windows/Linux or CMD + T on MacOS when writing a sketch in the Arduino IDE 2.0 will automatically format the sketch. It is possible to change the behaviour of this command. In this tutorial we will go through how you can change the behaviour of this command. 
+
+You can easily download the editor from the [Arduino Software page](https://www.arduino.cc/en/software#experimental-software). 
+
+You can also follow the [downloading and installing the Arduino IDE 2.0](/en/Tutorial/getting-started-with-ide-v2/ide-v2-downloading-and-installing) tutorial for more detailed guide on how to install the editor.
+
+## Requirements
+
+- Arduino IDE 2.0 installed. 
+
+## Setting the Custom Configuration
+
+It is possible to define your own custom configuration of the auto formatter feature in two different ways. The custom configuration of the auto formatter can be set on a global level to cover all sketches opened in the editor, or you can set the configuration to be specific to a sketch.
+
+The formatter tool used by Arduino IDE 2.0 is ClangFormat. The documentation for the configuration options is [here](https://clang.llvm.org/docs/ClangFormatStyleOptions.html).
+
+### Global scope
+If you add a `.clang-format` configuration file to either of the following locations, the Arduino IDE 2.0 will always use it instead of the Arduino default configuration.
+
+If you are using Windows place the file in:
+```
+C:\Users\<username>\.arduinoIDE\
+```
+Or
+```
+C:\Users\<username>\AppData\Local\Arduino15\
+```
+
+If you are using Linux place the file in:
+```
+~/.arduinoIDE/
+```
+Or
+```
+~/.arduino15/
+```
+
+If you are using macOS place the file in:
+```
+~/.arduinoIDE
+```
+Or
+```
+~/Library/Arduino15/
+```
+
+### Sketch scope
+If you add a `.clang-format` or `_clang-format` configuration file to the root of a sketch, the Arduino IDE will use that configuration when formatting that sketch. This file has precedence over a global formatter configuration file.
```

Comment:
```suggestion
If you add a `.clang-format` configuration file to the root of a sketch, the Arduino IDE will use that configuration when formatting that sketch. This file has precedence over a global formatter configuration file.
```

Even though the **ClangFormat** tool supports either filename if you use it directly, at this time, the only formatter configuration filename supported by Arduino IDE is `.clang-format`:

https://github.com/arduino/arduino-ide/blob/5ff9ce0028726a01d07511897150d07541a7b476/arduino-ide-extension/src/node/clang-formatter.ts#L110

I should have mentioned that limitation along with the quote I referenced from the ClangFormat documentation during my previous review.